vault/identity_store_entities.go

```diff
@@ -800,6 +862,28 @@ func (i *IdentityStore) mergeEntity(ctx context.Context, txn *memdb.Txn, toEntit
 		}
 
 		for _, alias := range fromEntity.Aliases {
+			if aliasId, ok := toEntityAccessors[alias.MountAccessor]; ok {
+				// Handle conflicts (conflict = both aliases share the same mount accessor)
+				if strutil.StrListContains(conflictingAliasIDsToKeep, aliasId) {
+					i.logger.Info("Deleting from_entity alias during entity merge", "from_entity", fromEntityID, "deleted_alias", alias.ID)
+					err := i.MemDBDeleteAliasByIDInTxn(txn, alias.ID, false)
+					if err != nil {
+						return nil, fmt.Errorf("failed to delete orphaned alias during merge: %w", err)
+					}
+
+					// Continue to next alias, as there's no alias to merge left in the from_entity
+					continue
+				} else if forceMergeAliases || strutil.StrListContains(conflictingAliasIDsToKeep, alias.ID) {
+					i.logger.Info("Deleting to_entity alias during entity merge", "to_entity", fromEntityID, "deleted_alias", aliasId)
```

Comment:
```suggestion
					i.logger.Info("Deleting to_entity alias during entity merge", "from_entity", fromEntityID, "deleted_alias", aliasId)
```

Reply:
Good catch, but the correct change is actually changing `fromEntityID` to `toEntity.ID`. I'll make the change to update alias names to make this a bit clearer.